lib/fluent/buffer.rb

```diff
@@ -1,336 +0,0 @@
-#
-# Fluentd
-#
-#    Licensed under the Apache License, Version 2.0 (the "License");
-#    you may not use this file except in compliance with the License.
-#    You may obtain a copy of the License at
-#
-#        http://www.apache.org/licenses/LICENSE-2.0
-#
-#    Unless required by applicable law or agreed to in writing, software
-#    distributed under the License is distributed on an "AS IS" BASIS,
-#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-#    See the License for the specific language governing permissions and
-#    limitations under the License.
-#
-module Fluent
-  class BufferError < StandardError
-  end
-
-  class BufferChunkLimitError < BufferError
-  end
-
-  class BufferQueueLimitError < BufferError
-  end
-
-
-  class Buffer
-    include Configurable
-
-    def initialize
-      super
-    end
-
-    def configure(conf)
-      super
-    end
-
-    def start
-    end
-
-    def shutdown
-    end
-
-    def before_shutdown(out)
-    end
-
-    def emit(key, data, chain)
-      raise NotImplementedError, "Implement this method in child class"
-    end
-
-    def keys
-      raise NotImplementedError, "Implement this method in child class"
-    end
-
-    def push(key)
-      raise NotImplementedError, "Implement this method in child class"
-    end
-
-    def pop(out)
-      raise NotImplementedError, "Implement this method in child class"
-    end
-
-    def clear!
-      raise NotImplementedError, "Implement this method in child class"
-    end
-  end
-
-
-  class BufferChunk
-    include MonitorMixin
-
-    def initialize(key)
-      super()
-      @key = key
-    end
-
-    attr_reader :key
-
-    def <<(data)
-      raise NotImplementedError, "Implement this method in child class"
-    end
-
-    def size
-      raise NotImplementedError, "Implement this method in child class"
-    end
-
-    def empty?
-      size == 0
-    end
-
-    def close
-      raise NotImplementedError, "Implement this method in child class"
-    end
-
-    def purge
-      raise NotImplementedError, "Implement this method in child class"
-    end
-
-    def read
-      raise NotImplementedError, "Implement this method in child class"
-    end
-
-    def open
-      raise NotImplementedError, "Implement this method in child class"
-    end
-
-    def write_to(io)
-      open {|i|
-        FileUtils.copy_stream(i, io)
-      }
-    end
-
-    def msgpack_each(&block)
-      open {|io|
-        u = MessagePack::Unpacker.new(io)
-        begin
-          u.each(&block)
-        rescue EOFError
-        end
-      }
-    end
-  end
-
-
-  class BasicBuffer < Buffer
```

Comment:
At `v0.14`, we will just use `Fluent::Plugin::Buffer` as a base class for all buffer plugins.